catalog/cataloger_diff.go

```diff
@@ -84,11 +85,15 @@ func (c *cataloger) Diff(ctx context.Context, repository string, leftReference s
 		return nil, false, fmt.Errorf("right reference: %w", err)
 	}
 
-	if limit < 0 || limit > DiffMaxLimit {
-		limit = DiffMaxLimit
```

Comment:
Nit: I prefer this type of logic: it's fewer lines and no `else`, also the common path is straight-line code and the `if` fixes an issue if there is ne.